src/main/webapp/mapsScript.js

```diff
@@ -90,6 +93,108 @@ function showDirections(locations) {
       window.alert('Directions request failed due to ' + status);
     }
   });
+
+  // Add markers to map.
+  addPoiMarker(origin, true, 0);  
+  for (let i = 0; i < waypts.length; i++) {
+    let index = i + 1;
+    addPoiMarker(waypts[i].location, false, index);
+  }
+}
+
+/* 
+ * Adds marker to map with appropriate letter label and an info window.
+ * location: String name or address of the POI.
+ * isOrigin: boolean indicating if the location is the origin (hotel) or not.
+             Used to determine the icon of the map marker.
+ * index: index of waypoint. Used to determine marker label.
+ */
+function addPoiMarker(location, isOrigin, index) {
+  // Query the location String to get Place details.
+  var service = new google.maps.places.PlacesService(map);
+  service.textSearch({
+    location: map.getCenter(),
+    radius: '500',
```

Comment:
This seems potentially fragile. We certainly don't expect anyone to enter a destination more than 500 (miles?) away, but it seems like they'd get confusing behavior if they did.

More broadly, it seems like doing a textSearch opens the door to various possible edge-cases like: matching the wrong location or matching multiple locations. I'm wondering if it's possible to add this functionality without a textSearch, eg: could you get the place ids from the TripDay and use those to add a map pin?

Reply:
The location/radius parameters in the search only bias the results to prefer locations in the circle, but it will still display locations outside the circle if found. Based on this, I would prefer to leave it in so that the text search is more likely to find the appropriate location we're referring to (e.g. Central Park in NY vs a diff Central Park). 

Ideally we would do one text search in the backend to get the place IDs and then only use the place IDs to refer to the locations from then on to reduce any chance of misrepresenting the location, but that would require more fundamental changes which are not priorities in the interest of time.

Reply:
That sounds good to me!